### PR TITLE
Glossarizer addEventListener

### PR DESCRIPTION
**Pull request by Miniland1333**
This PR relates a few issues involving the Glossarizer with the 'load' addEventListener

- [X] Bugfix buildBackMatter 'load' problems relating to `document.currentScript`
- [x] Work out a system so that a debug script can override a production script.
- [ ] Handle auto-activation of buildBackMatter within libretextsGlossarizer.js for simplicity

### CODE REVIEW

File: public/Leo Jayachandran/Glossarizer/libretextsGlossarizer.js [hunk 1]
```diff
@@ -667,9 +667,12 @@ class LibreTextsGlossarizer {
...
 }
 
+//enable debug mode
+LibreTexts.debug.glossarizer = true;
 
 //Self Initialize
 let libretextGlossary = new LibreTextsGlossarizer(); // Needs to be accessible to the sidebar buttons
-window.addEventListener('load', ()=>{
-    libretextGlossary.makeGlossary();
+window.addEventListener('load', () => {
```
Comment (Miniland1333) [inline public/Leo Jayachandran/Glossarizer/libretextsGlossarizer.js:675]: Add check maybe for document.currentScript src tag `includes('files.libretexts.org')`

**Comment by Miniland1333** [comment]
@Jonathanjaya, the next step if relevant would be to tackle point 2 of buildBackMatter automatically. If you want to skip this or save until later, you can.

Lastly, please test the modifications I've tried to make so that the debug override works. My test scripts from yesterday are located here: https://github.com/LibreTexts/Libretext/tree/master/public/Henry%20Agnew/debugOverride.

**Comment by Miniland1333** [comment]
@Jonathanjaya there are three things I would like you to check that I've done for this PR:
1. Modified how the glossarizer is initialized. Should allow for a development copy to automatically overwrite a production copy once this is deployed.
2. Added protections to BackMatter glossary like you already did for the regular glossary.
3. Strengthened Regex so that it is able to handle/ignore classes or other attributes in the column headers.

**Review by Jonathanjaya — APPROVED** [review]
Looks good, thanks for troubleshooting that for me.

I assume it was an extra class added to each cell.

**Pull request merged**